L1Trigger/L1TTwinMux/src/DTRPCBxCorrection.cc

```diff
@@ -0,0 +1,211 @@
+       bool shift_1 = false;
+        ///dtts exists and qual lt m_QualityLimit and (there is no primitve in the new bx or it was shifted or it is going to be removed as duplicate)
+      // if(dtts && dtts->code()<m_QualityLimit && (!dttsnew  || shifted[final_bx+3] || dups[final_bx+3])) {
+        if(dtts && dtts->code()<m_QualityLimit  && (!dttsnew  || shifted[final_bx+3] || dups[final_bx+3])) {
+          dtts_sh = new L1MuDTChambPhDigi( final_bx , dtts->whNum(), dtts->scNum(), dtts->stNum(),dtts->phi(), dtts->phiB(), dtts->code(), dtts->Ts2Tag(), dtts->BxCnt(),1);
+          l1ttma_outsh.push_back(*dtts_sh);
+          shifted[init_bx+3] = true;
+          shift_1 = true;}
+       if(dtts && dtts->code()<m_QualityLimit && dttsnew) dups[init_bx+3] = true;
+
+        //dtts exists and qual lt m_QualityLimit and dttsnew exists and the previous (shift_1) prim was not shifted and there is empty space in second TS
+       if(dtts && dtts->code()<m_QualityLimit && dttsnew && !shift_1 && !m_phiDTDigis_tm->chPhiSegm(wheel,station,sector,final_bx,flipBit(track_seg)) ) {
+        ///XXX: Source of discrepancies
+        ///in order to send as second TS the two prims must come from different halves of the station
+       ///this information does not exist in data
+       ///'simulate' this information by requiring different sign in phis and dphi>100
+           if(sign(dtts->phi())!=sign(dttsnew->phi())&& deltaPhi(-(dtts->phi()),dttsnew->phi())>100) {
+              dtts_sh2nd  = new L1MuDTChambPhDigi( final_bx , dtts->whNum(), dtts->scNum(), dtts->stNum(),dtts->phi(), dtts->phiB(), dtts->code(), flipBit(track_seg), dtts->BxCnt(),1);
+              secondTs[final_bx+3] = true;
+              dups[init_bx+3] = false;
+              shifted[init_bx+3] = true;
+              }
+          }
+        shiftedPhiDTDigis.setContainer(l1ttma_outsh);
+   }
+ }//end of bx
+
+       for(int bx=-3; bx<=3; bx++){
+         L1MuDTChambPhDigi * dtts=nullptr;
+         if(secondTs[bx+3] ) 
+           if(dtts_sh2nd) {m_l1ttma_out.push_back(*dtts_sh2nd); }
+         dtts = shiftedPhiDTDigis.chPhiSegm(wheel,station,sector,bx,track_seg);
+         if(dtts){m_l1ttma_out.push_back(*dtts); 
+                  continue;}
+         if(dups[bx+3]) continue;
+         ///if there is no shift then put the original primitive
+         dtts = m_phiDTDigis_tm->chPhiSegm(wheel,station,sector,bx,track_seg);
+         if(!shifted[bx+3] && dtts) {
+            m_l1ttma_out.push_back(*dtts);
+         }
+        }
+
+   }//end of station
+   }//end of sc
+  }//end of wheel
+}
+
+int DTRPCBxCorrection::deltaPhi(int dt_phi, int rpc2dt_phi ){
+  int delta_phi = abs( dt_phi - rpc2dt_phi );
+  return delta_phi;
+}
+
+std::vector<int> DTRPCBxCorrection::concat_delta(vector<int> delta_0, vector<int> delta_p, vector<int> delta_m){
+    vector<int> delta;
+    delta.insert(delta.end(), delta_0.begin(), delta_0.end());
```

Comment:
delta.push_back?
(after reserving 3)

Reply:
sorry, reserving the sum of the delta* lengths  (this is a rather expensive way to merge three vectors regardless)